aspnetcore/security/app-secrets.md

```diff
@@ -1,125 +1,250 @@
 
->[!WARNING]
-> Environment variables are generally stored in plain text and are not encrypted. If the machine or process is compromised, then environment variables can be accessed by untrusted parties. Additional measures to prevent disclosure of user secrets may still be required.
+[!code-csharp[](app-secrets/samples/1.1/UserSecrets/Startup.cs?name=snippet_StartupConstructor&highlight=10)]
+::: moniker-end
+
+Consider an ASP.NET Core web app in which **Individual User Accounts** security is enabled. A default database connection string is included in the project's *appsettings.json* file with the key `DefaultConnection`. The default connection string is for LocalDB, which runs in user mode and doesn't require a password. During app deployment, the `DefaultConnection` key value can be overridden with an environment variable's value. The environment variable may store the complete connection string with sensitive credentials.
+
+> [!WARNING]
+> Environment variables are generally stored in plain, unencrypted text. If the machine or process is compromised, environment variables can be accessed by untrusted parties. Additional measures to prevent disclosure of user secrets may be required.
 
 ## Secret Manager
 
-The Secret Manager tool stores sensitive data for development work outside of your project tree. The Secret Manager tool is a project tool that can be used to store secrets for a .NET Core project during development. With the Secret Manager tool, you can associate app secrets with a specific project and share them across multiple projects.
+The Secret Manager tool stores sensitive data during the development of an ASP.NET Core project. In this context, a piece of sensitive data is an app secret. App secrets are stored in a separate location from the project tree. The app secrets are associated with a specific project or shared across several projects. The app secrets aren't checked into source control.
 
->[!WARNING]
+> [!WARNING]
 > The Secret Manager tool doesn't encrypt the stored secrets and shouldn't be treated as a trusted store. It's for development purposes only. The keys and values are stored in a JSON configuration file in the user profile directory.
 
-## Installing the Secret Manager tool
+## How the Secret Manager tool works
 
-# [Visual Studio](#tab/visual-studio/)
+The Secret Manager tool abstracts away the implementation details, such as where and how the values are stored. You can use the tool without knowing these implementation details. The values are stored in a [JSON](https://json.org/) configuration file in a system-protected user profile folder on the local machine:
 
-Right-click the project in Solution Explorer, and select **Edit \<project_name\>.csproj** from the context menu. 
-Add the highlighted line to the *.csproj* file, and save to restore the associated NuGet package:
+* Windows: `%APPDATA%\Microsoft\UserSecrets\<user_secrets_id>\secrets.json`
```

Comment:
Originally, we wanted to provide tooling to work with this file. Using JSON to store the secrets was meant to be temporary until we could encrypt the secrets at rest. We *might* some day actually do that, which is why I wanted to express my reservations about documenting too much what goes on under the hood.

cc @glennc @danroth27 - thoughts on this? Originally, we said we wanted this file location to be an internal implementation detail that no one should ever see. But, two years in, I think enough people have discovered this that we could never change this without making a big breaking change.